addon_service/tasks/invocation.py

```diff
@@ -19,11 +23,18 @@ def perform_invocation__blocking(invocation: AddonOperationInvocation) -> None:
     # implemented as a sync function for django transactions
     with dibs(invocation):  # TODO: handle dibs errors
         try:
-            _imp = get_storage_addon_instance__blocking(
-                invocation.imp_cls,  # type: ignore[arg-type]  #(TODO: generic impstantiation)
-                invocation.thru_account,
-                invocation.storage_imp_config,
-            )
+            if isinstance(invocation.imp_cls, CitationAddonImp):
+                _imp = get_citation_addon_instance__blocking(
+                    invocation.imp_cls,  # type: ignore[arg-type]  #(TODO: generic impstantiation)
+                    invocation.thru_account,
+                    invocation.storage_imp_config,
+                )
+            else:
+                _imp = get_storage_addon_instance__blocking(
+                    invocation.imp_cls,  # type: ignore[arg-type]  #(TODO: generic impstantiation)
+                    invocation.thru_account,
+                    invocation.storage_imp_config,
+                )
```

Comment:
This may be not needed if multi-table inheritance proceeds in it's current form. So keep in mind, that you might need to remove this change when the time comes